src/metadata.ts

```diff
@@ -141,19 +152,25 @@ export class ApiObjectMetadataDefinition {
    */
   private readonly ownerReferences: OwnerReference[];
 
+  /**
+   * The ApiObject this metadata is attached to.
+   */
+  private readonly apiObject: ApiObject;
+
   /**
    * Additional metadata attributes passed through `options`.
    */
   private readonly _additionalAttributes: { [key: string]: any };
 
-  constructor(options: ApiObjectMetadata = { }) {
+  constructor(options: ApiObjectMetadataDefinitionOptions) {
```

Comment:
This is a breaking change, but I feel pretty confident no one (or very few) are directly using this API. It is mainly used in cdk8s itself.

Reply:
@iliapolo One of the very few checking in, this just led to some very confused debugging when it broke one of our libraries.